bentoml/yatai/client/deployment_api.py

```diff
@@ -103,123 +118,116 @@ def apply_deployment(self, deployment_info):
-                ] = operator_spec.get('num_of_gunicorn_workers_per_instance')
-        elif operator_value == DeploymentSpec.AWS_LAMBDA:
-            deployment_dict['spec']['aws_lambda_operator_config'] = {
-                'region': operator_spec.get('region')
-                or config().get('aws', 'default_region')
-            }
-            for field in ['api_name', 'memory_size', 'timeout']:
-                if operator_spec.get(field):
-                    deployment_dict['spec']['aws_lambda_operator_config'][
-                        field
-                    ] = operator_spec[field]
-        elif operator_value == DeploymentSpec.KUBERNETES:
-            deployment_dict['spec']['kubernetes_operator_config'] = {
-                'kube_namespace': operator_spec.get('kube_namespace', ''),
-                'replicas': operator_spec.get('replicas', 0),
-                'service_name': operator_spec.get('service_name', ''),
-                'service_type': operator_spec.get('service_type', ''),
-            }
-        else:
-            raise YataiDeploymentException(
-                'Platform "{}" is not supported in the current version of '
-                'BentoML'.format(platform)
+        deployment_pb = Deployment(
+            name=name, namespace=namespace, labels=labels, annotations=annotations
+        )
+        deployment_pb.spec.bento_name = bento_name
+        deployment_pb.spec.bento_version = bento_version
+        deployment_pb.spec.operator = DeploymentSpec.AWS_SAGEMAKER
+        deployment_pb.spec.sagemaker_operator_config.api_name = api_name
+        deployment_pb.spec.sagemaker_operator_config.instance_count = instance_count
+        deployment_pb.spec.sagemaker_operator_config.instance_type = instance_type
+        if region:
+            deployment_pb.spec.sagemaker_operator_config.region = region
+        if num_of_gunicorn_workers_per_instance:
+            deployment_pb.spec.sagemaker_operator_config.num_of_gunicorn_workers_per_instance = (  # noqa E501
+                num_of_gunicorn_workers_per_instance
             )
 
-        apply_response = self.apply(deployment_dict)
-
-        if apply_response.status.status_code == status_pb2.Status.OK:
-            describe_response = self.describe(deployment_name, namespace)
-            if describe_response.status.status_code == status_pb2.Status.OK:
-                deployment_state = describe_response.state
-                apply_response.deployment.state.CopyFrom(deployment_state)
-                return apply_response
-
+        apply_response = self.apply(deployment_pb)
+        if apply_response.status.status_code != status_pb2.Status.OK:
+            return apply_response
+        if wait:
+            state_result = self._wait_deployment_action_complete(name, namespace)
+            if state_result.status.status_code == status_pb2.Status.OK:
+                apply_response.deployment.state.CopyFrom(state_result.state)
```

Comment:
this looks a bit weird to me, let's just make the return type "GetDeploymentResponse", and always call GetDeployment at the end and return that response?